custom_functions/indicator_collect.py

```diff
@@ -52,37 +70,106 @@ def indicator_collect(container=None, **kwargs):
     elif isinstance(container, int):
         rest_container = phantom.requests.get(uri=phantom.build_phantom_rest_url('container', container), verify=False).json()
         if 'id' not in rest_container:
-            raise ValueError('Failed to find container with id {container}')
+            raise RuntimeError('Failed to find container with id {container}')
         container_dict = rest_container
         container_id = container
     else:
-        raise TypeError("The input 'container' is neither a container dictionary nor an int, so it cannot be used")
+        raise TypeError("The input 'container' is neither a container dictionary nor a valid container id, so it cannot be used")
+
+    if indicator_types_include:
+        indicator_types_include = [item.strip() for item in indicator_types_include.split(',')]
+    if indicator_types_exclude:
+        indicator_types_exclude = [item.strip() for item in indicator_types_exclude.split(',')]
+    if indicator_tags_include:
+        indicator_tags_include = [item.strip() for item in indicator_tags_include.split(',')]
+    if indicator_tags_exclude:
+        indicator_tags_exclude = [item.strip() for item in indicator_tags_exclude.split(',')]
+        
+    global_cef_mapping = phantom.requests.get(uri=phantom.build_phantom_rest_url('cef_metadata'), verify=False).json()['cef']
+    
+    if artifact_ids_include:
+        # Try to convert to a valid list
+        if isinstance(artifact_ids_include, str) and artifact_ids_include.startswith('[') and artifact_ids_include.endswith('['):
+            artifact_ids_include = json.loads(artifact_ids_include)
+        elif isinstance(artifact_ids_include, str):
+            artifact_ids_include = artifact_ids_include.replace(' ','').split(',')
+        
+        # Check validity of list
+        if not isinstance(artifact_ids_include, list) and not isinstance(artifact_ids_include, int):
+            raise TypeError(
+                f"Invalid artifact_ids_include entered: '{artifact_ids_include}'. Must be one of: json seriablize, comma separated, or an integer."
+            )
+        elif isinstance(artifact_ids_include, list) and not check_numeric_list(artifact_ids_include):
```

Comment:
If we want to accept a single integer artifact ID here, we need to check for an int and put it in a list.